front/src/store/index.js

```diff
@@ -50,11 +60,42 @@ export default new Vuex.Store({
     },
     logout ({ commit }) {
       commit('CLEAR_USER_DATA');
+    },
+    recoveryPasswordSendEmail ({ commit }, payload) {
+      return new Promise((resolve, reject) => {
+        axios.post(`${serverPath}api/ForgotPassword`,
+          { Email: payload.email }).then(resp => {
+          if (resp.status !== 200) {
+            reject(resp.data);
+          } else {
+            commit('SET_EMAIL_DATA', payload.email);
+            resolve();
+          }
+        });
+      });
+    },
+    recoveryPasswordSendPassword ({ commit, state }, passwordandtoken) {
+      return new Promise((resolve, reject) => {
+        axios.post(`${serverPath}api/ForgotPassword/ResetPassword`,
+          { Email: state.email, Password: passwordandtoken.password }, {
+            headers: { Authorization: 'Bearer ' + passwordandtoken.token }
+          }).then(resp => {
+          if (resp.data !== '') {
+            commit('CLEAR_EMAIL_DATA');
```

Comment:
seems this is not necessary as we don't want to store email data in frontend

Reply:
but why don't we want to store email on frontend? in your profile you will need it after logging in